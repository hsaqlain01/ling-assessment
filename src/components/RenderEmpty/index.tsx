import React from 'react';
import {View, Text} from 'react-native';
import tw from 'twrnc';

export const RenderEmpty = () => {
  return (
    <View style={tw`flex flex-1 justify-center items-center`}>
      <Text style={tw`text-black text-center text-lg w-80 mt-12`}>
        Please search for a user using their full name.
      </Text>
    </View>
  );
};
