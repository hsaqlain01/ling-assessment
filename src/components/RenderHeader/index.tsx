import React from 'react';
import {View, Text} from 'react-native';
import tw from 'twrnc';

const RenderHeader = () => {
  return (
    <View style={tw`flex-row w-full`}>
      <View
        style={tw`w-3/10 h-15 border-black items-center justify-center border-r-2 border-yellow-400  `}>
        <Text style={tw`my-3 text-black mt-5`}>Name</Text>
      </View>
      <View
        style={tw`w-2/10 h-15 border-black border-r-2   border-yellow-400  items-center justify-center`}>
        <Text style={tw`my-3 text-black  mt-7`}>Rank</Text>
      </View>
      <View
        style={tw`w-2/10  h-15 border-black border-r-2 items-center justify-center  border-yellow-400 `}>
        <Text style={tw`my-3 text-Black  mt-7`}>Bananas</Text>
      </View>
      <View
        style={tw`w-3/10 h-15 border-white items-center justify-center px-1 `}>
        <Text style={tw`my-3 text-black text-center text-xs mt-7`}>
          isSearchedUser?
        </Text>
      </View>
    </View>
  );
};

export default RenderHeader;
