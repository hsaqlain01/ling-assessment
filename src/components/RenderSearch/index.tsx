import {View, Text} from 'react-native';
import React from 'react';
import tw from 'twrnc';

import {IRenderData} from './types';

export const RenderSearch = ({user, search, index}: IRenderData) => {
  return (
    <View
      style={tw`flex-row w-full bg-white  border-t-2 h-15  ${
        index === 9 ? 'rounded-b-md border-b-2' : ''
      }  border-yellow-400  `}>
      <View
        style={tw`w-3/10 border-black items-center justify-center border-r-2 border-yellow-400  `}>
        <Text
          style={tw`my-3 ${
            search === user?.name ? 'text-red-700' : ' text-black'
          }  px-2 text-center `}>
          {user?.name}
        </Text>
      </View>
      <View
        style={tw`w-2/10 border-black border-r-2 items-center justify-center  border-yellow-400 `}>
        <Text style={tw`text-black`}>{user?.rank}</Text>
      </View>
      <View
        style={tw`w-2/10  border-black border-r-2  items-center justify-center  border-yellow-400 `}>
        <Text style={tw`text-black`}>{user?.bananas}</Text>
      </View>
      <View
        style={tw`w-3/10 border-white items-center justify-center  border-yellow-400 `}>
        <Text style={tw`text-black`}>
          {search === user?.name ? 'yes' : 'no'}
        </Text>
      </View>
    </View>
  );
};
