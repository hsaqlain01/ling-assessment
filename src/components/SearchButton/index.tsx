import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import tw from 'twrnc';

import {SearchButtonProps} from './types';

export const SearchButton = ({searchUser}: SearchButtonProps) => {
  return (
    <TouchableOpacity
      onPress={searchUser}
      style={tw`w-3/10 px-2 py-2 border-2 border-white rounded-xl items-center justify-center h-50px mt-5`}
      activeOpacity={0.5}>
      <Text style={tw`text-white font-semibold`}>Search</Text>
    </TouchableOpacity>
  );
};
