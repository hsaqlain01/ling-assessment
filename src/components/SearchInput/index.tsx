import React from 'react';
import {View, TextInput} from 'react-native';
import tw from 'twrnc';

import SearchIcon from '../../assets/svg/searchIcon.svg';
import {InputObject} from './types';

const SearchInput = ({setSearch, search, setFilteredData}: InputObject) => {
  return (
    <View
      style={tw`flex-row w-6.5/10 px-2 py-2 border-2 border-white rounded-xl h-50px mt-5`}>
      <View style={tw`mr-2`}>
        <SearchIcon fill={'#fff'} width="30" height="30" />
      </View>
      <TextInput
        value={search}
        onFocus={() => setFilteredData([])}
        style={tw`text-white font-semibold`}
        onChangeText={text => {
          text === ''
            ? (setFilteredData([]), setSearch(text))
            : setSearch(text);
        }}
        placeholder="Search"
        placeholderTextColor={'#fff'}
      />
    </View>
  );
};

export default SearchInput;
