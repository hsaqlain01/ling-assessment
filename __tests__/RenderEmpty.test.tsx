import React from 'react';
import renderer from 'react-test-renderer';
import RenderHeader from '../src/components/RenderHeader';
import {RenderEmpty} from '../src/components/RenderEmpty';

it('Header renders correctly', () => {
  const tree = renderer.create(<RenderEmpty />).toJSON();
  expect(tree).toMatchSnapshot();
});
