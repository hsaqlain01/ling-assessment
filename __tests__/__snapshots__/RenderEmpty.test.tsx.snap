// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#000",
        "fontSize": 18,
        "lineHeight": 28,
        "marginTop": 48,
        "textAlign": "center",
        "width": 320,
      }
    }
  >
    Please search for a user using their full name.
  </Text>
</View>
`;
